pkg/kubectl/drain/drain.go

```diff
@@ -39,14 +44,20 @@ const (
 type Helper struct {
 	Client              kubernetes.Interface
 	Force               bool
-	DryRun              bool
 	GracePeriodSeconds  int
 	IgnoreAllDaemonSets bool
 	Timeout             time.Duration
 	DeleteLocalData     bool
 	Selector            string
 	PodSelector         string
+	Out                 io.Writer
 	ErrOut              io.Writer
```

Comment:
Can we use `klogr` interface instead of a plain `io.Writer` here?

Reply:
klogr does not implement the io.Writer interface, so no.

Reply:
I'm suggesting to switch to `klogr` and remove the need of `io.Writer` within this library, given that's the logging facade used within k/k and subprojects.

Reply:
That seems like it would be out of scope for this patch set.

Reply:
@justinsb wdyt? 

/cc @neolit123

Reply:
SGTM, but klogr is still not used in k/k. so this needs +1s from some groups.

klog is now owner by sig-instrumentation technically, but also it's up to sig-cli if they like the idea.
given @justinsb as an approver for klog is already in the loop, feel free to poke more folks from this OWNERS file (e.g. thockin):
https://github.com/kubernetes/klog/blob/master/OWNERS

on the sig-cli side
/cc @soltysh

Reply:
I think it's a solid idea (though I really want to see klogr output that is beautiful).  But can we keep this PR as a refactor?

I guess the challenge is that we're effectively making io.Writer into a contract?  I don't think we have API guarantees here, so I think we could change it later.  (As always, we shouldn't just do so for fun, but this feels legitimate.)

I do think the two things we should work on in klogr (and forgive me if these already exist) are "plaintext output" (for people that just want a simple CLI tool) and "pretty structured output" (to actually get some value out of the structured logging!

Reply:
(And I actually think if we had an adapter from io.Writer -> trivial klogr we could then absolutely change this to klogr)